.github/workflows/build.yaml

```diff
@@ -26,11 +28,17 @@ jobs:
           key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
           restore-keys: |
             ${{ runner.os }}-gradle-
+      - name: Get Node.js version
+        run: |
+          echo "$(node -v)" > node-version.txt
+      - name: Get npm version
+        run: |
+          echo "$(npm -v)" > npm-version.txt
       - name: Cache the NPM Cache
         uses: actions/cache@v2
         with:
           path: ~/.npm
-          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          key: ${{ runner.os }}-node-${{ hashFiles('node-version.txt', 'npm-version.txt', '**/package-lock.json') }}
```

Comment:
Fix issue with `actions/cache@v2` not noticing that the cached npm files are not valid as they were created using a different version of node and/or npm